crates/core/tedge/src/cli/certificate/create.rs

```diff
@@ -47,6 +58,19 @@ impl CreateCertCmd {
         self.create_test_certificate_for(config, &KeyKind::Reuse { keypair_pem })
     }
 
+    pub fn create_certificate_signing_request(
+        &self,
+        config: &NewCertificateConfig,
+    ) -> Result<(), CertError> {
+        // Reuse private key if it already exists
+        let key_kind = match std::fs::read_to_string(&self.key_path) {
+            Ok(keypair_pem) => KeyKind::Reuse { keypair_pem },
+            Err(err) if err.kind() == ErrorKind::NotFound => KeyKind::New,
+            Err(err) => return Err(CertError::IoError(err).cert_context(self.cert_path.clone())),
+        };
+        self.create_test_certificate_for(config, &key_kind)
```

Comment:
Okayish. The method name is a bit confusing. It's not obvious that this will create a CSR because the config provides some path to store it.